tests/plugins/find_and_replace.test.ts

```diff
@@ -290,163 +430,343 @@ describe("basic search", () => {
     deleteFilter(model, "A1:A6");
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
   });
 
   test("Need to maintain search if column or row is added", () => {
     model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
     let matches = model.getters.getSearchMatches();
     let matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
     addRows(model, "after", 1, 1);
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
+    expect(matches).toHaveLength(3);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 4, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 5, selected: false });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 4, selected: false });
   });
 });
+
 describe("next/previous cycle", () => {
   beforeEach(() => {
-    model = new Model({ sheets: [{ id: "s1" }] });
+    const sheet1 = "s1";
+    model = new Model({ sheets: [{ id: sheet1 }] });
     setCellContent(model, "A1", "1");
     setCellContent(model, "A2", "1");
-    setCellContent(model, "A3", "1");
+    const sheet2 = "s2";
+    createSheet(model, { sheetId: sheet2 });
+    setCellContent(model, "A3", "1", sheet2);
+    searchOptions = {
+      matchCase: false,
+      exactMatch: false,
+      searchFormulas: false,
+      searchScope: "activeSheet",
+      specificRange: { _sheetId: sheet2, _zone: toZone("A1:D13") },
+    };
```

Comment:
create a DEFAULT_SEARCH_OPTION constant if you're about to repeat the same thing for the 4th beforeEach